src/cli.js

```diff
@@ -167,120 +166,97 @@ function checkContributors(argv) {
-        // const contributor = {
-        //   login: usr.login,
-        //   contributions: [],
-        // }
-        // console.log('commit auth:', usr)
-        const existingContributor = contributorsToAdd.filter(
-          ctrb => ctrb.login === usr.login,
+      if (!guessedCategory) {
+        console.warn(
+          `Oops, I couldn't find any category for the "${lbl}" label`,
         )
-        if (existingContributor.length) {
-          //there's no label or commit message info so use only code for now
-          if (!existingContributor[0].contributions.includes('code')) {
-            existingContributor[0].contributions.push('code')
-          }
-        } else
-          contributorsToAdd.push({login: usr.login, contributions: ['code']})
-      })
 
-      // console.log('contributorsToAdd=', contributorsToAdd)
-      contributorsToAdd.forEach(contributor => {
+        return
+      }
+
+      if (!contributor.contributions.includes(guessedCategory)) {
+        contributor.contributions.push(guessedCategory)
+
         console.log(
-          `Adding ${chalk.blue(contributor.login)} for ${chalk.underline(
-            contributor.contributions.join('/'),
+          `Adding ${chalk.blue(usr.login)} for ${chalk.underline(
+            guessedCategory,
           )}`,
         )
-        args._ = ['', contributor.login, contributor.contributions.join(',')]
-        // if (contributor.contributions.length) addContribution(args)
-        // else console.log('Skipping', contributor.login)
-      })
-    },
-    err => console.error('fetch error:', err),
-  )
+      }
+    })
+
+    const existingContributor = contributorsToAdd.find(
+      c => c.login === usr.login,
+    )
+
+    if (existingContributor) {
+      existingContributor.contributions.push(...contributor.contributions)
+    } else {
+      contributorsToAdd.push(contributor)
+    }
+  })
+
+  commitAuthors.forEach(usr => {
+    const existingContributor = contributorsToAdd.find(
+      c => c.login === usr.login,
+    )
+
+    if (existingContributor) {
+      // There's no label or commit message info so use only code for now
+      if (!existingContributor.contributions.includes('code')) {
+        existingContributor.contributions.push('code')
+      }
+    } else {
+      contributorsToAdd.push({login: usr.login, contributions: ['code']})
+    }
+  })
+
+  // TODO: Roll onto other contribution categories following https://www.draw.io/#G1uL9saIuZl3rj8sOo9xsLOPByAe28qhwa
+
+  for (const contributor of contributorsToAdd) {
+    if (!contributor.contributions.length) {
+      console.log('Skipping', contributor.login)
+
+      return
+    }
+
+    console.log(
+      `Adding ${chalk.blue(contributor.login)} for ${chalk.underline(
+        contributor.contributions.join('/'),
+      )}`,
+    )
+
+    args._ = ['', contributor.login, contributor.contributions.join(',')]
+
+    /* eslint-disable no-await-in-loop */
+    await addContribution(args)
```

Comment:
Wouldn't you think that having error handling would be a good idea here?

Reply:
What should be the behaviour? Return a console log message? kill the import?

Reply:
Having a `console.error` would the minimum I believe although there's situation where `throw`ing an error would be better depending on where it happens.
I doubt having the whole process being cancelled if it fails at adding a specific type of contribution or a contributor in a specific list (`console.log`/`console.warn` would be more appropriate there as the process isn't atomic, unless it being so would be better?).
However, for the `await addContribution(args)`, something like:
```suggestion
try {
  /* eslint-disable no-await-in-loop */
  await addContribution(args);
} catch (err) {
  console.error(err);
}
```
Or something of that sort so users can see the error and help us debug and fix the error in case there's a failure.
What do you think?

Reply:
Makes sense. I wrote something like this:

```
console.log(
        `Adding ${chalk.blue(contributor.login)} for ${chalk.underline(
          contributions,
        )} Failed: ${JSON.stringify(error)}`,
      )
```